pkg/pipelines/scm/gitlab.go

```diff
@@ -0,0 +1,122 @@
+package scm
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/openshift/odo/pkg/pipelines/meta"
+	"github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	triggersv1 "github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+)
+
+const (
+	gitlabPRBindingName   = "gitlab-pr-binding"
+	gitlabPushBindingName = "gitlab-push-binding"
+
+	gitlabCIDryRunFilters = "( header.match ( ‘X-Gitlab-Event’ , ‘Merge Request Hook’ ) && body.object_kind == ‘merge_request’ ) && body.object_attributes.state == ‘opened’ && body.project.path_with_namespace == %s  && body.project.default_branch == body.object_attributes.target_branch )"
+
+	gitlabCDDeployFilters = "( header.match ( ‘X-Gitlab-Event’ , ‘Push Hook’ ) && body.object_kind == ‘push’ && body.project.path_with_namespace == %s && body.ref.EndsWith (body.project.default_branch )"
```

Comment:
Probably don't need the `body.object_kind == 'push'` bit, if we're receiving Push Hooks that are _not_ `object_kind == 'push'` then something is broken elsewhere.

This `body.project.path_with_namespace == %s`  doesn't look like it's quoting the name of the value.

And it doesn't look like it's quoting the strings correctly `’` doesn't look like a string quote?